models/classes/runner/time/TimerAdjustmentMapInterface.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoTests\models\runner\time;
+
+
+/**
+ * Interface TimerAdjustmentMapInterface
+ *
+ * Describes an API for timer adjustment storage and access.
+ *
+ * @package oat\taoTests\models\classes\runner\time
+ */
+interface TimerAdjustmentMapInterface
+{
+    public const ACTION_INCREASE = 'increase';
+    public const ACTION_DECREASE = 'decrease';
+
+    /**
+     * Puts an entry to the map
+     * @param string $sourceId
+     * @param string $action
+     * @param int $seconds
+     * @return TimerAdjustmentMapInterface
+     */
+    public function put(string $sourceId, string $action, int $seconds): TimerAdjustmentMapInterface;
```

Comment:
why don't we use to separate methods: increase(sourceId, seconds), decrease(sourceId, seconds)

Reply:
because it's a `map` data structure and I think it's a better method name to describe the action of adding a new value

Reply:
but we don't need to know how it works inside correct? we don't need the put action, we only need to increase or decrease value, and how it will be implemented inside doesn't have matter